src/mixins/serviceHandler.js

```diff
@@ -55,6 +57,8 @@ export default {
         console.error(`[${error?.response?.status}] Something went wrong, Please try again later!`)
 
         return Promise.reject(error)
+      } finally {
+        this.isLoading = false
```

Comment:
This is not how you should handle `isLoading` using Polkadot.

I assume you are doing this because your cancel order loading button is not processing correctly, you shouldn't set `isLoading` to false inside a `finally` because it will resolve before events are fulfilled by the blockchain. You should set `isLoading` inside the callback function I made.

Your code will break other functionalities in production. For example, I ran your code in my development environment and the loading symbol no longer waits until the event finishes in the blockchain.

Please refer to this commit: https://github.com/debionetwork/debio-frontend/commit/17dbe1d0e67b1de941a08e182f893648acf4d0d5

Please read this doc of PolkadotJS: https://polkadot.js.org/docs/api/start/api.query.subs/#subscriptions-with-params

And finally, please remove this. If you have questions or other proposals please reach me via Discord.

Reply:
Oh and if you're wondering WHY cancel order callback doesn't work, it's because the callback hasn't been set inside the `polkadotProvider`s please check the difference between these two files:
- https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/orders.js
- https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/labs/index.js

These two files use PolkadotJS, but all `labs` commands use a callback while most `orders` commands do not. Anyway, you can even check with the `fulfillOrder` command inside `orders` here: https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/orders.js#L8

Reply:
The solution? Make a separate universal callback function similar to the one you see here: https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/labs/index.js#L27

Does it seem repetitive? Yes, it does. You are open to suggest inputs if you like. But so far, this is how you should do it.

Reply:
Hi @agustinustheo, I have a question. Does that mean all requests have to use callbacks?

Reply:
All commands must use callbacks @bukandicki, albeit not all of them will use them for the same reason.h